src/Compilers/CSharp/Test/Emit2/Semantics/ExperimentalAttributeTests.cs

```diff
@@ -0,0 +1,1000 @@
+        comp.VerifyDiagnostics(
+            // 0.cs(1,1): warning DiagID1: 'C' is for evaluation purposes only and is subject to change or removal in future updates.
+            // C.M();
+            Diagnostic("DiagID1", "C").WithArguments("C").WithLocation(1, 1)
+            );
+
+        var diag = comp.GetDiagnostics().Single();
+        Assert.Equal("DiagID1", diag.Id);
+        Assert.Equal(ErrorCode.WRN_Experimental, (ErrorCode)diag.Code);
+        Assert.Equal(DefaultHelpLinkUri, diag.Descriptor.HelpLinkUri);
+    }
+
+    [Theory, CombinatorialData]
+    public void EmptyUrlFormat(bool inSource)
+    {
+        var libSrc = """
+[System.Diagnostics.CodeAnalysis.Experimental("DiagID1", UrlFormat = "")]
+public class C
+{
+    public static void M() { }
+}
+""";
+
+        var src = """
+C.M();
+""";
+
+        var comp = inSource
+            ? CreateCompilation(new[] { src, libSrc, experimentalAttributeSrc })
+            : CreateCompilation(src, references: new[] { CreateCompilation(new[] { libSrc, experimentalAttributeSrc }).EmitToImageReference() });
+
+        comp.VerifyDiagnostics(
+            // (1,1): warning DiagID1: 'C' is for evaluation purposes only and is subject to change or removal in future updates.
+            // C.M();
+            Diagnostic("DiagID1", "C").WithArguments("C").WithLocation(1, 1)
+            );
+
+        var diag = comp.GetDiagnostics().Single();
+        Assert.Equal("DiagID1", diag.Id);
+        Assert.Equal(ErrorCode.WRN_Experimental, (ErrorCode)diag.Code);
+        Assert.Equal("", diag.Descriptor.HelpLinkUri);
+    }
+
+    [Fact]
+    public void NullUrlFormat()
+    {
+        // We use a default help URL if the UrlFormat is improper
+        var src = """
+C.M();
+
+[System.Diagnostics.CodeAnalysis.Experimental("DiagID1", UrlFormat = null)]
+class C
+{
+    public static void M() { }
+}
+""";
+        var comp = CreateCompilation(new[] { src, experimentalAttributeSrc });
+        comp.VerifyDiagnostics(
+            // 0.cs(1,1): warning DiagID1: 'C' is for evaluation purposes only and is subject to change or removal in future updates.
+            // C.M();
+            Diagnostic("DiagID1", "C").WithArguments("C").WithLocation(1, 1)
+            );
+
+        var diag = comp.GetDiagnostics().Single();
+        Assert.Equal("DiagID1", diag.Id);
+        Assert.Equal(ErrorCode.WRN_Experimental, (ErrorCode)diag.Code);
+        Assert.Equal(DefaultHelpLinkUri, diag.Descriptor.HelpLinkUri);
+    }
+
+    [Fact]
+    public void FullyQualified()
+    {
+        var src = """
+N.C.M();
+
+namespace N
+{
+    [System.Diagnostics.CodeAnalysis.Experimental("DiagID1")]
+    class C
+    {
+        public static void M() { }
+    }
+}
+""";
+        var comp = CreateCompilation(new[] { src, experimentalAttributeSrc });
+        comp.VerifyDiagnostics(
+            // 0.cs(1,1): warning DiagID1: 'C' is for evaluation purposes only and is subject to change or removal in future updates.
```

Comment:
>'C' [](http://example.com/codeflow?start=43&length=3)

The name may not be fully-qualified. I believe the `Diagnostic(...)` item below may be using a different display format than the comment, and the comment appears to match what is reported in command-line errors.

Reply:
Indeed. Thanks for catching that